File: leetcode/2788. Split Strings by Separator.py
#* 2788. Split Strings by Separator
words = ["one.two.three","four.five","six"]
separator = "."
from typing import List
class Solution:
    def splitWordsBySeparator(self, words: List[str], separator: str) -> List[str]:
        return [word for segment in words for word in segment.split(separator) if word != '']
    
print(Solution().splitWordsBySeparator(words,separator))
        

File: leetcode/2788. Split Strings by Separator.java
// 2788. Split Strings by Separator
//words = ["one.two.three","four.five","six"]
//separator = "."
import java.util.List;
import java.util.ArrayList;

class Solution {
    public List<String> splitWordsBySeparator(List<String> words, char separator) {
        List<String> result = new ArrayList<>();
        for (String word : words) {
            String[] splitWords = word.split(Character.toString(separator));
            for (String splitWord : splitWords) {
                if (!splitWord.isEmpty()) {
                    result.add(splitWord);
                }
            }
        }
        return result;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        List<String> words = List.of("one.two.three", "four.five", "six");
        char separator = '.';
        List<String> result = solution.splitWordsBySeparator(words, separator);
        System.out.println(result);
    }
}
